protocols/direct-funding_test.go

```diff
@@ -1,14 +1,88 @@
 package protocols
```

Comment:
something to consider here: there are two options for module here. you can use:
1. "protocols"
2. "protocols_test"

Which you choose has a subtle effect on your tests. Both will get dropped from compiled code. However, the second will mean you are in a different module, which means you'll only be able to access public vars/types. Sometimes this can be useful to make sure you're only testing public interfaces. (which may force important decisions). Other times, it may be really hard to test public interfaces without having access to a couple private vars. There's no right answer -- just a helpful language feature to know about.

Reply:
This is great, thanks. Go compiler makes really liberal use of magic strings (`init` functions, `main` package, `internal` packages, etc) and this is yet-another example I hadn't been aware of.

Will check these comments out more thoroughly tomorrow.

Reply:
I would lean towards `protocols_test` to place the constraints on the tests. Sounds we can switch at the drop of a hat with 1LOC change in case that adds too much of a burden.

Switching in the other direction is more burdensome unless we are adhering to the "only test public interfaces" rule. And if we are, why not lean on the compiler?